Content.Shared/Gibbing/Systems/GibbingSystem.cs

```diff
@@ -0,0 +1,309 @@
+    /// <param name="gibContentsOption">What type of gibing do we perform on any container contents?</param>
+    /// <param name="droppedEntities">a hashset containing all the entities that have been dropped/created</param>
+    /// <param name="gibable">The gibable component</param>
+    /// <param name="randomSpreadMod">How much to multiply the random spread on dropped giblets(if we are dropping them!)</param>
+    /// <param name="playAudio">Should we play audio</param>
+    /// <param name="allowedContainers">A list of containerIds on the target that permit gibing</param>
+    /// <param name="excludedContainers">A list of containerIds on the target that DO NOT permit gibing</param>
+    /// <param name="launchCone">The cone we are launching giblets in (if we are launching them!)</param>
+    /// <param name="launchGibs">Should we launch giblets or just drop them</param>
+    /// <param name="launchDirection">The direction to launch giblets (if we are launching them!)</param>
+    /// <param name="launchImpulse">The impluse to launch giblets at(if we are launching them!)</param>
+    /// <param name="launchImpulseVariance">The variation in giblet launch impulse (if we are launching them!)</param>
+    /// <param name="logMissingGibable">Should we log if we are missing a gibbableComp when we call this function</param>
+    /// <returns>True if successful, false if not</returns>
+    public bool TryGibEntityWithRef(EntityUid outerEntity, EntityUid target, GibType gibType, GibContentsOption gibContentsOption,
+        ref HashSet<EntityUid> droppedEntities, GibbableComponent? gibable = null, bool launchGibs = true,
+        Vector2? launchDirection = null, float launchImpulse = 0f, float launchImpulseVariance = 0f, Angle launchCone = default,
+        float randomSpreadMod = 1.0f, bool playAudio = true, List<string>? allowedContainers = null,
+        List<string>? excludedContainers = null, bool logMissingGibable = false)
+    {
+        if (!Resolve(target, ref gibable, logMissing: false))
+        {
+            DropEntity(target, Transform(outerEntity), randomSpreadMod, null, ref droppedEntities,
+                launchGibs, launchDirection, launchImpulse, launchImpulseVariance, launchCone);
+            if (logMissingGibable)
+            {
+                Log.Warning($"{ToPrettyString(target)} does not have a GibbableComponent! " +
+                            $"This is not required but may cause issues contained items to not be dropped.");
+            }
+
+            return false;
+        }
+
+        if (gibType == GibType.Skip && gibContentsOption == GibContentsOption.Skip)
+            return true;
+        if (launchGibs)
+        {
+            randomSpreadMod = 0;
+        }
+        var parentXform = Transform(outerEntity);
+        HashSet<BaseContainer> validContainers = new();
+        foreach (var container in _containerSystem.GetAllContainers(target))
```

Comment:
Funny container names are not relevant for body part gibbing, since this method should only be called on body parts. If there is an issue with a particular container, rather than hardcode it, you can specify containers to exclude by name in the gibentity arguments